tube.py.sample

```diff
@@ -5,19 +5,26 @@ Automatically run tests when files change by running: stir
 See: https://github.com/thomasw/testtube
 """
 
-from testtube.helpers import Helper, Flake8, Pep257
+from testtube.helpers import Flake8, Helper, Nosetests, Pep257
 
 
-class Clear(Helper):
-    """Clear the screen."""
-
+class ScreenClearer(Helper):
     command = 'clear'
 
+    def success(self, *args):
+        pass
+
+
+class FailFirstNosetests(Nosetests):
+    def get_args(self):
+        return ['-x']
 
-clear = Clear(all_files=True)
+clear = ScreenClearer(all_files=True)
 lint = Flake8(all_files=True)
-docstrings = Pep257(all_files=True)
+lint_docs = Pep257(all_files=True)
+test = FailFirstNosetests()
 
 PATTERNS = (
-    (r'.*\.py$', [clear, lint, docstrings], {'fail_fast': True}),
+    (r'.*\.py$', [clear, test], {'fail_fast': True}),
+    (r'.*\.py$', [lint, lint_docs]),
```

Comment:
You don't have to change this, but generally I like to lint before tests because they run quickly and find problems like missing imports, typos, etc that will cause the (slower) tests to fail anyway.